File: RayTracerModelVB/SurfaceTexture.vb
Public Class SurfaceTexture

    Public Property Diffuse As Func(Of Vector3, Color)

    Public Property Specular As Func(Of Vector3, Color)

    Public Property Reflect As Func(Of Vector3, Double)

    Public Property Roughness As Double

    Public Sub New(ByVal diffuse As Func(Of Vector3, Color), ByVal specular As Func(Of Vector3, Color), ByVal reflect As Func(Of Vector3, Double), ByVal roughness As Double)
        Me.Diffuse = diffuse
        Me.Specular = specular
        Me.Reflect = reflect
        Me.Roughness = roughness
    End Sub
End Class

File: RayTracerModel/SurfaceTexture.cs
using System;

namespace RayTracer
{
    public class SurfaceTexture
    {
        public Func<Vector3, Color> Diffuse { get; private set; }
        public Func<Vector3, Color> Specular { get; private set; }
        public Func<Vector3, double> Reflect { get; private set; }
        public double Roughness { get; private set; }

        public SurfaceTexture(Func<Vector3, Color> diffuse, 
            Func<Vector3, Color> specular,
            Func<Vector3, double> reflect,
            double roughness)
        {
            Diffuse = diffuse;
            Specular = specular;
            Reflect = reflect;
            Roughness = roughness;
        }
    }
}
